pyogrio/_err.pyx

```diff
@@ -248,3 +344,197 @@ cdef void error_handler(CPLErr err_class, int err_no, const char* err_msg) nogil
 
 def _register_error_handler():
     CPLPushErrorHandler(<CPLErrorHandler>error_handler)
+
+cpl_errs = GDALErrCtxManager()
+
+
+cdef class StackChecker:
```

Comment:
```suggestion
cdef class ErrorHandler:
```

The stack part seems an implementation detail, I think the part we want to focus here on the name for usage below is that it checks for errors.